environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java

```diff
@@ -246,4 +253,30 @@ public ValidationResult validateStorageLocation(String storageLocation, String s
         }
         return resultBuilder.build();
     }
+
+    public ValidationResult validateEncryptionKeyUrl(EnvironmentCreationDto creationDto, EntitlementService entitlementService) {
+        ValidationResultBuilder resultBuilder = ValidationResult.builder();
+        if (AZURE.name().equalsIgnoreCase(creationDto.getCloudPlatform())) {
+            String encryptionKeyUrl = getEncryptionKeyUrlIfPresent(creationDto);
+            if (StringUtils.isNotEmpty(encryptionKeyUrl)) {
+                if (!entitlementService.isAzureDiskSSEWithCMKEnabled(creationDto.getAccountId())) {
+                    resultBuilder.error(String.format("You have specified encryption-key-url to enable Server Side Encryption for Azure Managed disks with CMK"
+                            + "but that feature is currently not enabled for this account."
+                            + " Please get 'CDP_CB_AZURE_DISK_SSE_WITH_CMK' enabled for this account to use SSE with CMK."));
+                } else {
+                    ValidationResult validationResult = encryptionKeyUrlValidator.validateEncryptionKeyUrl(encryptionKeyUrl);
+                    resultBuilder.merge(validationResult);
+                }
+            }
+        }
+        return resultBuilder.build();
+    }
+
+    private String getEncryptionKeyUrlIfPresent(EnvironmentCreationDto creationDto) {
+        try {
+            return creationDto.getParameters().getAzureParametersDto().getAzureResourceEncryptionParametersDto().getEncryptionKeyUrl();
+        } catch (NullPointerException e) {
+            return null;
+        }
```

Comment:
No, please reformulate this. I suggest `Optional.ofNullable(creationDto.getParameters()).map(getAzureParametersDto()).map(getAzureResourceEncryptionParametersDto()).map(getEncryptionKeyUrl()).orElse(null)`.